fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MvUtils.java

```diff
@@ -1455,4 +1469,70 @@ public static SlotRef extractPartitionSlotRef(Expr paritionExpr) {
         Preconditions.checkState(slotRefs.size() == 1);
         return slotRefs.get(0);
     }
+
+    /**
+     * Inactive related mvs after modified columns have been done. Only inactive mvs after
+     * modified columns have done because the modified process may be failed and in this situation
+     * should not inactive mvs then.
+     */
+    public static void inactiveRelatedMaterializedViews(Database db,
+                                                        OlapTable olapTable,
+                                                        Set<String> modifiedColumns) {
+        if (modifiedColumns == null || modifiedColumns.isEmpty()) {
+            return;
+        }
+        // inactive related asynchronous mvs
+        for (MvId mvId : olapTable.getRelatedMaterializedViews()) {
+            MaterializedView mv = (MaterializedView) db.getTable(mvId.getId());
+            if (mv == null) {
+                LOG.warn("Ignore materialized view {} does not exists", mvId);
+                continue;
+
+            }
+            // TODO: support more types for base table's schema change.
+            try {
+                MvPlanContext mvPlanContext = MvPlanContextBuilder.getPlanContext(mv);
+                if (mvPlanContext != null) {
+                    OptExpression mvPlan = mvPlanContext.getLogicalPlan();
+                    List<ColumnRefOperator> usedColRefs = MvUtils.collectScanColumn(mvPlan, scan -> {
+                        if (scan == null) {
+                            return false;
+                        }
+                        Table table = scan.getTable();
+                        return table.getId() == olapTable.getId();
+                    });
+                    Set<String> usedColNames = usedColRefs.stream().map(x -> x.getName()).collect(Collectors.toSet());
+                    for (String modifiedColumn : modifiedColumns) {
+                        if (usedColNames.contains(modifiedColumn)) {
```

Comment:
should we consider the case of column name?

Reply:
Good question, fix it.